src/features/selectedPost/selectedPostSlice.ts

```diff
@@ -0,0 +1,15 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+
+const selectedPostSlice = createSlice({
+  name: 'selectedPost',
+  initialState: null as Post | null,
+  reducers: {
+    setSelectedPost: (state, action: PayloadAction<Post | null>) => {
+      state = action.payload;
```

Comment:
The assignment `state = action.payload;` does not mutate the state correctly in Redux Toolkit. Instead, you should directly return the new state or use `return action.payload;` to update the state. Redux Toolkit uses Immer under the hood, which allows direct mutation of the state object, but assigning to `state` directly does not work as expected.